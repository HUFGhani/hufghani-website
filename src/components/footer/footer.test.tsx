import { render } from '@testing-library/react'
import React from 'react'
import Footer from '.'
import { SEOContext } from '../../contexts/seoContext'

const testdata = {
  author: 'Hamza Umar Farooq Ghani',
}

function renderFooter() {
  return render(
    <SEOContext.Provider value={testdata}>
      <Footer />
    </SEOContext.Provider>
  )
}

describe('<Footer/>', () => {
  it('should render Footer', () => {
    const { asFragment } = renderFooter()
    expect(asFragment()).toMatchSnapshot()
  })
  it('should render the copyright text correctly', () => {
    const user = { name: 'Giorgio' }
    const { getByTestId } = renderFooter()
    expect(getByTestId('copyRight-year')).toHaveTextContent(
      `Â© 2017 - ${new Date().getFullYear()}, Hamza Umar Farooq Ghani`
    )
  })
})
