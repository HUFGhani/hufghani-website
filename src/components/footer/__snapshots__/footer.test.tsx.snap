// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer/> should render Footer 1`] = `
<DocumentFragment>
  <footer
    class="css-12pe2qd"
  >
    <p
      aria-label="© 2017 - 2021, Hamza Umar Farooq Ghani"
      class="chakra-text css-0"
      data-testid="copyRight-year"
    >
      © 2017 - 2021, Hamza Umar Farooq Ghani
    </p>
  </footer>
</DocumentFragment>
`;
